### PR TITLE
Crud endpoints garden

### PR DESCRIPTION
**Pull request by D-Lessenden**
Create CRUD endpoints for gardens. For sad-path testing I followed Hash's lead for the CRUD user endpoints and return a 204 No Content.

### CODE REVIEW

File: app/serializers/garden_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class GardenSerializer
+  include FastJsonapi::ObjectSerializer
+  attributes :latitude
+  attributes :longitude
+  attributes :name
+end
```
Comment (hashmaster3k) [inline app/serializers/garden_serializer.rb:6]: These can be done on one line separated by commas but it isn't a problem of any sorts.

File: app/controllers/api/v1/gardens_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  def destroy
+    return nil if !Garden.exists?(params[:id])
+    Garden.destroy(params[:id])
+  end
+
+  private
+  def garden_params
+    params.permit(:latitude, :longitude, :name, :description)
+  end
+end
```
Comment (hashmaster3k) [inline app/controllers/api/v1/gardens_controller.rb:31]: Controller looks great! Organized and light.

File: spec/requests/api/v1/garden_request_spec.rb [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      end
+
+      it 'can destroy a garden' do
+        garden = create(:garden)
+
+        expect(Garden.count).to eq(1)
+        delete "/api/v1/gardens/#{garden.id}"
+        expect(response).to be_successful
+        expect(Garden.count).to eq(0)
+        expect{Garden.find(garden.id)}.to raise_error(ActiveRecord::RecordNotFound)
```
Comment (hashmaster3k) [inline spec/requests/api/v1/garden_request_spec.rb:106]: Good check on before and after destroying a garden to ensure the garden was created in the first place.

File: spec/requests/api/v1/garden_request_spec.rb [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+     expect(response.status).to eq(204)
+   end
+
+   it 'delete - returns a 204 if query entered wrong' do
+     delete "/api/v1/gardens/999999"
+     expect(response).to be_successful
+     expect(response.status).to eq(204)
+   end
+  end
+end
```
Comment (hashmaster3k) [inline spec/requests/api/v1/garden_request_spec.rb:141]: I had a brief conversation with @AngelaGuardia about returning a 204 No Content. While I think it's a great idea to do sad-path testing and account for incorrect data being sent, I believe later we should revise the controller to send back the proper HTTP code based on the [JSON Specification Guide](https://jsonapi.org/format/). Good for now I think.

**Review by hashmaster3k — APPROVED** [review]
I pulled down the branch and everything seems to be working well. You have 100% coverage. I also noticed that .exits? is a built-in rails method. That is awesome because in my User CRUD endpoints I made the method...oh well.

File: app/serializers/garden_serializer.rb (hunk 1 above)
Comment (D-Lessenden) [reply app/serializers/garden_serializer.rb:6]: That is good to know for the future. Thanks.

File: spec/requests/api/v1/garden_request_spec.rb (hunk 4 above)
Comment (D-Lessenden) [reply spec/requests/api/v1/garden_request_spec.rb:141]: Do you think we should make an extension or another issue that explicitly deals with the 204 error?

**Pull request merged**